engine/render/proto/render/compute_program_ddf.proto

```diff
@@ -4,10 +4,14 @@ package dmRenderDDF;
 import "ddf/ddf_extensions.proto";
 import "ddf/ddf_math.proto";
 
+import "render/material_ddf.proto";
+
 option java_package = "com.dynamo.render.proto";
 option java_outer_classname = "ComputeProgram";
 
 message ComputeProgramDesc
 {
-    required string program = 1 [(resource)=true];
+    required string program                  = 1 [(resource)=true];
+    repeated MaterialDesc.Constant constants = 2;
+    repeated MaterialDesc.Sampler samplers   = 3;
```

Comment:
It's a little bit annoying that the sampler and constant descs are embedded in the material, but oh well.